README.md

```diff
@@ -1,19 +1,230 @@
+Also works by executing the file directly:
+
+```console
+$ cd microenginewebhookspy
+$ ./engine.py analyze --help
+Usage: eicar-sample analyze [OPTIONS] [ARTIFACTS]...
+
+  Analyze artifacts
+
+Options:
+  -v, --verbose
+  --check-empty     Verify this engine can analyze an empty
+                                  bounty
+  --check-eicar     Verify this engine can analyze EICAR test
+                    file
+  --check-wicar, --check-exploit-url
+                    Verify this engine can analyze the WICAR
+                    exploit kit URL
+  -t, --artifact-type [bounty|file|url]
+                    Artifact type to use when constructing
+                    bounties. 'bounty' loads manually
+                    constructed bounties, treating each argument
+                    as the path to a JSON-encoded bounty object
+--help              Show this message and exit.
+```
+
+The returned value will be checked for structure.
+
+This CLI can issue scans for files in your disk, for local testing pourposes:
+
+```console
+$ ./engine.py analyze ~/Downloads/Firefox\ Installer.exe
+{
+  ...
+  "verdict": "benign",
+  "bid": 999900000000000000
+}
+```
+
+We recommend that you always check scans for:
+- EMPTY bounties
+- EICAR if creating a file-scanning engines
+- WICAR if creating a url-scanning engine
+- Return UNKNOWN for unsupported file types
+
+## Example: Checking the file type
+
+If you run an analysis for WICAR the template implementation will return BENIGN:
+
+```console
+$ ./engine.py analyze --check-wicar
+{
+  ...
+  "verdict": "benign",
+  "bid": 999900000000000000
+}
+...
+AssertionError: Received 'benign' instead of malicious
+```
+
+As an example, for handling URL bounties gracefully,
+you can change the `engine.py` file to have this new lines:
+
+```diff
+ @engine.register_analyzer
+ def analyze(bounty: psengine.Bounty) -> psengine.Analysis:
++    if not psengine.bounty.is_file_artifact(bounty):
++        log.error("Received non-file artifact bounty")
++        return psengine.bounty.UNSUPPORTED
+     contents = psengine.get_artifact_bytes(bounty)
+```
+
+It will now change to answer non-file bounties with an UNSUPPORTED verdict.
+
+```console
+$ ./engine.py analyze --check-wicar
+2025-10-22 20:30:19,022 - ERROR [engine.py:28][analyze] Received non-file artifact bounty
+{
+  ...
+  "verdict": "unknown",
+  "bid": 0
+}
+...
+AssertionError: Received 'unknown' instead of malicious
+```
+
+Which is fine for an EICAR engine, that is not supposed to handle URL bounties.
+
+# Where to go from here?
+
+This simple engine now does everything in the correct way.
+Your existing malware-detection tool can be wired inside `engine.py` freely.
+
+Some nice tooling exists inside `psengine` package. For example, if your tool
+can natively scan files running in the filesystem via CLI, there is a
+context manager function that downloads the file and stores in a temporary
+folder on disk, easing your life:
```

Comment:
```suggestion
An existing malware-detection tool can be freely integrated within `engine.py`.

Helpful tooling is also available in the `psengine` package. For instance, if the
tool can natively scan files on the filesystem via CLI, a context manager
function is provided that downloads the file to a temporary folder on disk,
simplifying the process:
```

Reply:
Here again, I wanted to be explicit that _you, that hava a malware-detection tool, can make it work here_.

Had also avoided fancy workding on purpose, but I may had stretched the tone.

Please confirm and I can accept it.